src/components/TodosContext/Contexts.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {
+  ReactNode,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  useLocation,
+} from 'react-router-dom';
+// import { useLocation } from 'react-router-dom';
+import { Todo } from '../../types/Todo';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import { TypeTodosContext } from '../../types/TypeTodosContext';
+import { Status } from '../../types/Status';
+
+type Props = {
+  children: ReactNode;
+};
+
+export const TodosContext = React.createContext<TypeTodosContext>({
+  todos: [],
+  setTodos: () => { },
+});
+
+export const VisibleTodosContext = React.createContext<Todo[]>([]);
+export const SelectedContext = React.createContext(Status.All);
+
+export const TodosContextProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
```

Comment:
it can be the simple var instead of the state

Reply:
I don’t understand how visibleTodos can be a simple variable. It is used in other components. But I grouped them into 1 context.